app.js

```diff
@@ -1 +1,100 @@
- 
\ No newline at end of file
+// Book Array to hold books
+const myLibrary = [
+  {
+    title: 'Default One',
+    author: 'John Doe',
+    pages: 1000,
+    read: false,
+  }, {
+    title: 'Default Two',
+    author: 'Jane Doe',
+    pages: 1500,
+    read: false,
+  }];
+
+// Book function
+function Book(title, author, pages, read) {
+  this.title = title;
+  this.author = author;
+  this.pages = pages;
+  this.read = read;
+}
+
+// Function to add books to View
+function addBookToLibrary(book) {
+  const list = document.querySelector('#book-table');
+  const row = document.createElement('tr');
+  book.read = document.getElementById('checkbox').checked;
+  const state = book.read === true ? 'Yes I have' : 'No I have not';
+
+  row.innerHTML = `
+    <td>${book.title}</td>
+    <td>${book.author}</td>
+    <td>${book.pages}</td>
+    <td><button class="state-btn" id="status">${state}</button></td>
+    <td><a href="#" class="del-btn delete">Delete Book</a></td>
+  `;
+  list.appendChild(row);
+
+  document.querySelector('body').addEventListener('click', (e) => {
+    if (e.target.id === 'status') {
+      e.preventDefault();
+      if (book.read === false) {
+        book.read = true;
+        e.target.innerHTML = 'Yes I have';
+      } else {
+        book.read = false;
+        e.target.innerHTML = 'No I have not';
+      }
+    }
+  });
+}
+
+// Function to clear fields after book is submitted
+function clearfields() {
```

Comment:
- [ ] [OPTIONAL] While you have separated the functions between logic and DOM manipulation, it would still be desirable to organize them a bit more, either putting all of the logic at the first half of the file or making two files, one for the logic and another for the DOM manipulation. If you go for the second option remember to include the extra script tag in your HTML file.